MSI: Use common types. Fix definitions for proper .NET SDK generation.

angosms opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

angosms committed:
Use common types. Fix definitions for proper .NET SDK generation.

AutorestCI commented:
/azp run automation - sdk

allenjzhang commented on specification/msi/resource-manager/Microsoft.ManagedIdentity/stable/2018-11-30/ManagedIdentity.json:
```diff
@@ -254,7 +254,7 @@
             "required": true,
             "schema": {
-              "$ref": "#/definitions/IdentityPatch"
+              "$ref": "#/definitions/IdentityUpdate"
```
This is a breaking change on an existing stable API-version and should be reverted. [See error details](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=257293&view=logs&j=b6885ff5-5b63-57e7-07bc-9d8d90d734f1&t=982b7c5c-6d4b-53e9-42d9-af8a8061e6e5&l=3093) 

You should skip modifying stable version of API.

angosms replied:
This was just added in this: https://github.com/Azure/azure-rest-api-specs/pull/7528

I saw that other specs had operationXUpdate and wanted to stay consistent with most of Azure.

allenjzhang replied:
As you can see that .NET SDK automation failed due to breaking changes by this. This kind of change should really be avoided. And I apologize for missing the previous rename in the 7528. Can you please change it back to **Identity**?

allenjzhang replied:
Discussed offline. Needs to split the model because of PATCH support.

allenjzhang commented on specification/msi/resource-manager/Microsoft.ManagedIdentity/preview/2015-08-31-preview/ManagedIdentity.json:
```diff
@@ -385,54 +369,39 @@
       "allOf": [
         {
-          "$ref": "#/definitions/Resource"
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/Resource"
```
nit: Would TrackedResource be what you are looking for? It is Resource + location + tags. Does have an extra field tags.

angosms replied:
> These api-version all seems go way back 2 - 5 years. If they are widely used, I would suggest if you want to hold off and apply the correct reference in the next API-version.

It seems like our swagger spec had minor issues and did not reflect the exact behavior of our API so this PR and the previous were addressing these issues. The referencefixes are there to fix the .NET SDK generation.

angosms replied:
I'll double check that on our side.

allenjzhang commented:
These api-version all seems go way back 2 - 5 years. If they are widely used, I would suggest if you want to hold off and apply the correct reference in the next API-version.

angosms commented:
> These api-version all seems go way back 2 - 5 years. If they are widely used, I would suggest if you want to hold off and apply the correct reference in the next API-version.

It seems like our swagger spec had minor issues and did not reflect the exact behavior of our API so this PR and the previous were addressing these issues. The referencefixes are there to fix the .NET SDK generation.

angosms committed:
Use tracked and proxy resource types

AutorestCI commented:
/azp run automation - sdk

angosms committed:
Revert reference of IdentityPatch to Identity resource

AutorestCI commented:
/azp run automation - sdk

angosms committed:
Revert the previous revert

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?
